packages/node_modules/@webex/plugin-meetings/src/roap/handler.js

```diff
@@ -0,0 +1,166 @@
+            }
+          })
+          .catch((err) => {
+            config.console.info('Cannot reset the peer connection', err);
+          });
+        return true;
+      }
+    }
+    if (!RoapUtil.ensureMeeting(meeting, action.type)) {
+      return true;
+    }
+  }
+  return false;
+};
+
+const handleSessionStep = (roap, session) => {
+  const {seq: sequenceId, messageType} = roap.msg;
+  if (session.OFFER && messageType === OFFER) {
+    session.GLARE_OFFER = roap.msg;
+    session.GLARE_OFFER.remote = !!roap.remote;
+    console.info(`Its a GLARE condition seq:${sequenceId} new mercury event`);
+  }
+  else {
+    console.info(
+      `Save OFFER/ANSWER seq:${sequenceId} new mercury event ${messageType}local state: ${JSON.stringify(
+        session.state.state,
+        null,
+        2
+      )}`
+    );
+    session[messageType] = roap.msg;
+    session[messageType].remote = !!roap.remote;
+  }
+};
+
+export default class RoapHandler extends StatelessSparkPlugin {
+  perform(session, meeting, action) {
+    switch (session.state.state) {
+      // case ROAP.ROAP_STATE.INIT:
+      // case ROAP.ROAP_STATE.WAIT_RX_OFFER:
+      // case ROAP.ROAP_STATE.WAIT_RX_ANSWER:
+      // case ROAP.ROAP_STATE.WAIT_RX_OK:
+      // TODO: default audio and video , change later
+      case ROAP.ROAP_STATE.WAIT_TX_ANSWER:
+        // TODO: sometime the you get an answer while you are creating an offer so SKIP
+        // Server will send the mercury event comes back
+        if (RoapUtil.shouldHandleMedia(meeting)) {
+          RoapUtil.updatePeerConnection(meeting, session)
+            .then((answerSdps) => {
+              Events.emit(EVENTS.ROAP_ANSWER, {
+                locusId: meeting.locusId,
+                locusSelfId: meeting.locus.self.id,
+                mediaId: meeting.mediaId,
+                sdps: answerSdps,
+                roapSeq: session.OFFER.seq,
+                correlationId: meeting.correlationId
+              });
+            })
+            .catch((err) => {
+              console.log(err);
+            });
+        }
+        break;
+      case ROAP.ROAP_STATE.WAIT_TX_OK:
+        RoapUtil.setRemoteDescription(meeting, session).then((res) => {
+          Events.emit(EVENTS.ROAP_OK, res);
+        });
+        break;
+      // case ROAP.ROAP_STATE.IDLE_LOCAL_OFFER:
+      case ROAP.ROAP_STATE.ERROR:
+        config.console.error('Call-RoapHandler#ERROR state=error', session);
+        break;
+      case ROAP.ROAP_STATE.GLARE:
+        session.GLARE_OFFER.tieBreaker = session.GLARE_OFFER.tieBreaker || 0;
+        session.OFFER.tieBreaker = session.OFFER.tieBreaker || 0;
+        config.console.error('Resolve the GLARE condition');
+        if (session.GLARE_OFFER.tieBreaker < session.OFFER.tieBreaker) {
+          // 2
+          config.console.error('local offer wins');
+        }
+        else {
+          config.console.error('remote offer wins', session);
```

Comment:
We'll want to change these to use `this.logger.error` but I figure we'll take a big pass of all the files and do that at a later point

Reply:
yes